src/app/search/components/selected-filters/SelectedFilters.tsx

```diff
@@ -0,0 +1,58 @@
+"use client";
+
+import React, { useContext } from "react";
+
+import { FiltersContext } from "@/app/search/providers/FiltersProvider";
+
+import Card from "@/components/card/Card";
+import { FiltersType } from "@/types/filters-type";
+import SolarTrashBinMinimalistic2Bold from "@/icons/SolarTrashBinMinimalistic2Bold";
+
+import styles from "./SelectedFilters.module.css";
+
+function SelectedFilters() {
+  const { dispatchFilters, filters } = useContext(FiltersContext);
+
+  const deleteAllClickHandler = (): void => {
+    dispatchFilters({ type: "deletedAllFilters" });
+  };
+
+  const deleteClickHandler = (key: keyof FiltersType): void => {
+    dispatchFilters({ type: "deletedFilter", key });
+  };
+
+  const keyValues: (keyof FiltersType)[] = Object.keys(
+    filters,
+  ) as (keyof FiltersType)[];
```

Comment:
اسم بهتر استفاده کنید
نیازی به این همه Explicit Type نیست، یه دونه کافیه

```suggestion
  const filterKeys = Object.keys(filters) as (keyof FiltersType)[];
```